File: src/main/java/com/exampledigisphere/vitapetcare/admin/timePeriod/domain/TimePeriod.java
package com.exampledigisphere.vitapetcare.admin.timePeriod.domain;

import com.exampledigisphere.vitapetcare.admin.workDay.domain.WorkDay;
import com.exampledigisphere.vitapetcare.config.root.BaseEntity;
import com.exampledigisphere.vitapetcare.config.root.Info;
import jakarta.persistence.*;

import java.time.LocalTime;

@Entity
@Table(name = "wrk_times_periods")
@Info(
  dev = Info.Dev.heltonOliveira,
  label = Info.Label.doc,
  date = "29/12/2025",
  description = "Entidade que representa um período de tempo dentro de um dia de trabalho"
)
public class TimePeriod extends BaseEntity<TimePeriod> {

  @Transient
  public static final String TABLE_NAME = "wrk_times_periods";
  private LocalTime startTime;
  private LocalTime endTime;
  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "work_day_id")
  private WorkDay workDay;

  public TimePeriod() {
  }

  public LocalTime getStartTime() {
    return startTime;
  }

  public void setStartTime(LocalTime startTime) {
    this.startTime = startTime;
  }

  public LocalTime getEndTime() {
    return endTime;
  }

  public void setEndTime(LocalTime endTime) {
    this.endTime = endTime;
  }

  public WorkDay getWorkDay() {
    return workDay;
  }

  public void setWorkDay(WorkDay workDay) {
    this.workDay = workDay;
  }

  public interface Authority {
    String TIME_PERIOD_CREATE = "TIME_PERIOD_CREATE";
    String TIME_PERIOD_EDIT = "TIME_PERIOD_EDIT";
    String TIME_PERIOD_VIEW = "TIME_PERIOD_VIEW";
    String TIME_PERIOD_VIEW_LIST = "TIME_PERIOD_VIEW_LIST";
    String TIME_PERIOD_DELETE = "TIME_PERIOD_DELETE";
  }
}

File: src/main/kotlin/com/exampledigisphere/vitapetcare/admin/timePeriod/domain/TimePeriod.kt
package com.exampledigisphere.vitapetcare.admin.timePeriod.domain

import com.exampledigisphere.vitapetcare.admin.workDay.domain.WorkDay
import com.exampledigisphere.vitapetcare.config.root.BaseEntity
import com.exampledigisphere.vitapetcare.config.security.Permissions
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonView
import jakarta.persistence.*
import org.hibernate.Hibernate
import org.jetbrains.annotations.NotNull
import java.time.LocalTime

@Entity
@Table(name = "wrk_times_periods")
class TimePeriod : BaseEntity() {

  @field:Transient
  final val TABLE_NAME = "wrk_times_periods";

  @field:NotNull
  @field:JsonView(Json.List::class)
  lateinit var startTime: LocalTime;

  @field:NotNull
  @field:JsonView(Json.List::class)
  lateinit var endTime: LocalTime;

  @field:NotNull
  @field:ManyToOne(fetch = FetchType.LAZY)
  @field:JoinColumn(name = "work_day_id")
  @field:JsonView(Json.WithWorkDay::class)
  @field:JsonIgnoreProperties("shifts")
  lateinit var workDay: WorkDay;

  fun loadWorkDay() = Hibernate.initialize(workDay);

  interface Json {
    interface List;
    interface Detail : List;
    interface WithWorkDay : WorkDay.Json.List;
    interface All : Detail, WithWorkDay;
  }

  object TimePeriodPermissions : Permissions {
    override var resource = "TIME_PERIOD";
  }
}
